src/common/LogClient.cc

```diff
@@ -165,6 +165,8 @@ void LogChannel::update_config(map<string,string> &log_to_monitors,
   int graylog_port = atoi(graylog_port_str.c_str());
 
   set_log_to_monitors(to_monitors);
+  if (!to_monitors)
```

Comment:
i'd suggest create two commits:

1. move `LogChannel::set_log_to_monitors(bool v)` to .cc file 
2. add a method to `LogClient`, something like `LogClient::reset()`, which clears `log_queue` and sync `last_log_sent` with `last_log`. and accompany this method with comment explaining that it avoid the hole in the disabling-then-enabling case. yes, `log_queue.clear()` would be a no-op when we re-enable `log_to_monitor`, but simpler this way.
3. in `LogChannel::set_log_to_monitors()` if `log_to_monitors` is different from the new `v`, in addition to assigning `v` to `log_to_monitors`, call `parent->reset()`.

for two reason:

1. better performance, we don't need to acquire the lock twice for getting the seq and for syncing `last_log_sent` when `log_to_monitors` is disabled
2. simpler this way.

also, please always add a space after `if`. and add curly brackets around the if block even if there is only a single statement in it, for better readability.

Reply:
also, could you prefix the title of your commit message with the subcomponent your are changing ? see https://github.com/ceph/ceph/blob/master/SubmittingPatches.rst#3-describe-your-changes